### PR TITLE
Add gitSync optional env description

### PR DESCRIPTION
**Pull request by 1yangsh**
When fetching a remote git repository to a local directory,
optional command to be executed after syncing a new hash of the remote repository as occasion demands.

Added descriptions of several environmental variables for script execution.

`GIT_SYNC_PERMISSION`
- Change permissions on the checked-out files to the specified mode.

`GIT_SYNC_EXECHOOK_BACKOFF` (default: 3s)
- The time to wait before retrying a failed --exechook-command.

`GIT_SYNC_EXECHOOK_COMMAND` (script to be executed)
- An optional command can be executed after syncing a new hash of the remote repository.

`GIT_SYNC_EXECHOOK_TIMEOUT` (default: 30s)
- The timeout for the --exechook-command.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by potiuk — APPROVED** [review]

File: chart/values.yaml [hunk 1]
```diff
@@ -1919,6 +1919,19 @@ dags:
...
+     #   value: "0755"
+     # The time to wait before retrying a failed --exechook-command.
+     # - name: GIT_SYNC_EXECHOOK_BACKOFF
+     #   value: "3s"
+     # An optional command to be executed after syncing a new hash of the remote repository.
+     # - name: GIT_SYNC_EXECHOOK_COMMAND
+     #   value: "./scripts/entrypoint.sh"
+     # The timeout for the --exechook-command.
+     # - name: GIT_SYNC_EXECHOOK_TIMEOUT
+     #   value: "30s"
```
Comment (jedcunningham) [inline chart/values.yaml]:
```suggestion
     # Supported env vars for gitsync can be found at https://github.com/kubernetes/git-sync
     # - name: ""
     #   value: ""
     
```

Let's be general here, I don't want to end up duplicating the whole gitsync docs here.

**Review by jedcunningham — CHANGES_REQUESTED** [review]

File: chart/values.yaml (hunk 1 above)
Comment (1yangsh) [reply chart/values.yaml]: I was thinking of expressing only the frequently used env values, but it would be better to give a general guide.

**Review by jedcunningham — APPROVED** [review]

File: chart/values.yaml [hunk 2]
```diff
@@ -1942,6 +1942,10 @@ dags:
 
     extraVolumeMounts: []
     env: []
+     # Supported env vars for gitsync can be found at https://github.com/kubernetes/git-sync
+     # - name: ""
+     #   value: ""
```
Comment (jedcunningham) [inline chart/values.yaml]:
```suggestion
    # Supported env vars for gitsync can be found at https://github.com/kubernetes/git-sync
    # - name: ""
    #   value: ""
```

**Review by jedcunningham — APPROVED** [review]

**Comment by jedcunningham** [comment]
Thanks @1yangsh! Congrats on your first commit 🎉

**Pull request merged**